hip-clang.docker

```diff
@@ -3,12 +3,13 @@ FROM ubuntu:18.04
 ARG PREFIX=/usr/local
 ARG GPU_ARCH=";"
 ARG MIOTENSILE_VER="default"
+ARG USE_TARGETID="OFF"
 
 # Support multiarch
 RUN dpkg --add-architecture i386
 
 # Add rocm repository
-RUN sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/rocm/apt/.apt_3.7/ xenial main > /etc/apt/sources.list.d/rocm.list'
+RUN if [ "$USE_TARGETID" = "ON" ] ; then sh -c 'echo deb [arch=amd64 trusted=yes] http://compute-artifactory.amd.com/artifactory/list/rocm-osdb-deb/ compute-rocm-dkms-no-npi-hipclang 6416 > /etc/apt/sources.list.d/rocm.list'; else sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/rocm/apt/.apt_3.7/ xenial main > /etc/apt/sources.list.d/rocm.list'; fi
```

Comment:
Please notice that all the tensile related test rebuild the docker with image arg of:
image: image+'-hip-clang-targetid

Reply:
Noticed. But `-hip-clang` is already included into image "basename", as you can see above. That is why my review comments suggest replacing `image+'-hip-clang...'` with `image+'...' `.

Reply:
You PR will replace `-hip-clang` with simple `-v2`. This is Ok, because we are getting rid of non-hip-clang based dockers, so there is no need to specifically mention hip-clang in image names.

Reply:
My PR is actually use the name `image+'-hip-clang-targetid`, which is requested by our former team leader. Personally I'm OK with any name as long as we don't have to change it every other month. Is this `-v2` a temporal or long term solution? I think maybe we can change the bothering but not malfunctioning names in another PR.

Reply:
Actually all this is not required, see #801 (#806 implements it).